src/types/track-modal.predicate.ts

```diff
@@ -0,0 +1,15 @@
+import {CreateEditModalData, DeleteTrackModalData} from "./track-modal.type";
+import { TrackSearchItem } from './track-api.type';
+
+export function isTrackData(data: DeleteTrackModalData): data is { track: TrackSearchItem } {
+  return !!data.track && !data.tracks;
+}
+
+export function isTracksData(data: DeleteTrackModalData): data is { tracks: string[] } {
+  return !!data.tracks && !data.track;
+}
+
+export function isTrackDataDefined(data: CreateEditModalData): data is TrackSearchItem {
+  return !!data;
```

Comment:
add additional check for key fields:

return !!data && typeof data === 'object' && 'id' in data && 'title' in data;

it is much better to use